src/mtev_conf.c

```diff
@@ -445,7 +445,24 @@ void mtev_override_console_stopword(int (*f)(const char *)) {
 
 static char *root_node_name = NULL;
 static char master_config_file[PATH_MAX] = "";
-static xmlXPathContextPtr xpath_ctxt = NULL;
+static pthread_key_t xpath_ctxt_key;
+struct xpath_ctxt_gen {
+  uint64_t gen;
+  xmlXPathContextPtr xpath;
+};
+static uint64_t master_ctxt_gen = 0;
+static __thread struct xpath_ctxt_gen xpath_ctxt_gen;
+static xmlXPathContextPtr master_xpath_ctxt(void) {
+  if(xpath_ctxt_gen.gen != ck_pr_load_64(&master_ctxt_gen)) {
+    if(xpath_ctxt_gen.xpath) xmlXPathFreeContext(xpath_ctxt_gen.xpath);
+    xpath_ctxt_gen.xpath = NULL;
+  }
+  if(xpath_ctxt_gen.xpath == NULL) {
+    if(master_config)
+      xpath_ctxt_gen.xpath = xmlXPathNewContext(master_config);
+  }
+  return xpath_ctxt_gen.xpath;
+}
```

Comment:
What happens if master_config is NULL in L461? Should we assert master_config? ... or is that a legit state.